internal/provider/data_source_http_test.go

```diff
@@ -216,48 +220,178 @@ func TestDataSource_UpgradeFromVersion2_2_0(t *testing.T) {
 	})
 }
 
+func TestDataSource_WithCACertificate(t *testing.T) {
+	t.Parallel()
+	svr := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	}))
+	defer svr.Close()
+
+	resource.UnitTest(t, resource.TestCase{
+		ProtoV6ProviderFactories: protoV6ProviderFactories(),
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(`
+							data "http" "http_test" {
+  								url = "%s"
+
+  								ca_certificate = <<EOF
```

Comment:
Nitpick: not strictly necessary, but it would be nice to make use of the TLS provider, and generate a CA certificate on the fly, instead of having to carry a fixture.

Reply:
Feels overcomplicated. Using the [testcert](https://pkg.go.dev/net/http/internal/testcert) certificates from golang package should sufficient for this test case.